src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,75 @@
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import mate.academy.springboot.datajpa.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.mapper.ProductMapper;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@AllArgsConstructor
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+    private final CategoryService categoryService;
+
+    @PostMapping
+    public ProductResponseDto create(@RequestBody ProductRequestDto requestDto) {
+        Product newProduct = productService.save(productMapper.mapToModel(requestDto));
+        return productMapper.mapToDto(newProduct);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return productMapper.mapToDto(productService.getByID(id));
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteById(@PathVariable Long id) {
+        productService.deleteBy(id);
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id,
+            @RequestBody ProductRequestDto requestDto) {
+        Product product = productMapper.mapToModel(requestDto);
+        product.setId(id);
+        return productMapper.mapToDto(productService.save(product));
+    }
+
+    @GetMapping("/price")
+    public List<ProductResponseDto> getAllByPriceBetween(@RequestParam BigDecimal from,
+                                                         @RequestParam BigDecimal to) {
+        return productService.getAllByPriceBetween(from, to)
+                .stream()
+                .map(productMapper::mapToDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/categories")
+    public List<ProductResponseDto> getAllByCategoriesIn(@RequestParam List<Long> ids) {
+        List<Category> categories = ids.stream()
+                .map(categoryService::getById)
+                .collect(Collectors.toList());
+        List<Product> products = productService.getAllByCategoriesIn(categories);
```

Comment:
you can use your ids directly in the service and repository, don't fetch categories by id